src/zif_abapgit_definitions.intf.abap

```diff
@@ -172,9 +172,12 @@ INTERFACE zif_abapgit_definitions PUBLIC.
       type    TYPE zif_abapgit_definitions=>ty_type,
       data    TYPE xstring,
       adler32 TYPE ty_adler32,
+      index   TYPE i,
     END OF ty_object .
   TYPES:
-    ty_objects_tt TYPE STANDARD TABLE OF ty_object WITH DEFAULT KEY .
+    ty_objects_tt TYPE STANDARD TABLE OF ty_object WITH DEFAULT KEY
+      WITH NON-UNIQUE SORTED KEY sha COMPONENTS sha1 type index
+      WITH NON-UNIQUE SORTED KEY type COMPONENTS type sha1 index .
```

Comment:
Components are
`sha1 type index`
and
`type sha1 index `

it looks like the index component is not used? if this is the case then one of the other fields can also be removed so it becomes

`sha1`
and
`type sha1`

Reply:
You are correct. Originally I had a unique index and thus specified all key components, but in testing I quickly realized that a non-unique key was the way to go. I simply never removed the excess fields.
Theindex field is only used to to restore the original sort order after working on the objects with the secondary indices.